tfhe/src/integer/u256.rs

```diff
@@ -9,6 +9,90 @@ pub const fn adc(l: u64, r: u64, c: bool) -> (u64, bool) {
 #[derive(Default, Copy, Clone, Debug, PartialEq, Eq)]
 pub struct U256(pub(crate) [u64; 4]);
 
+impl U256 {
+    /// Replaces the current value by interpreting the bytes in big endian order
+    pub fn copy_from_be_byte_slice(&mut self, bytes: &[u8]) {
+        // We internally have
+        //
+        // [ index 0    |  index 1   |  index 2 | index 3     ]
+        // [WB0,.., WB7 | WB0,..,WB7 | WB0,..,WB7 | WB0,..,WB7]
+        // [B0,.., B7   | B8 ,..,B15 | B16,..,B23 | B24,..,B31]
+        //   Least                                     Most
+        //
+        // Where each [WB0,...WB7] are in target_endian order,
+        // so, if target_endian == big it means, the full range
+        // B0..B31 is not a true little endian order,
+        // thus requiring an additional step
+
+        let inner_slice = self.0.as_mut_slice();
+        inner_slice.fill(0);
+        unsafe {
+            // SAFETY SLICES are contiguous
+            // (4 * 64) bits / 8 bits = 32
+            let inner_byte_slice =
+                std::slice::from_raw_parts_mut(inner_slice.as_mut_ptr() as *mut u8, 32);
+            let shortest_len = inner_byte_slice.len().min(bytes.len());
+            inner_byte_slice[..shortest_len].copy_from_slice(&bytes[..shortest_len]);
+            inner_byte_slice.reverse();
+        }
+        #[cfg(target_endian = "big")]
+        for word in self.0.iter_mut() {
+            *word = word.swap_bytes();
+        }
+    }
+
+    /// Replaces the current value by interpreting the bytes in little endian order
+    pub fn copy_from_le_byte_slice(&mut self, bytes: &[u8]) {
+        // Same principle as in copy_from_be_byte_slice applies here
+
+        let inner_slice = self.0.as_mut_slice();
+        inner_slice.fill(0);
+        unsafe {
+            // SAFETY SLICES are contiguous
+            // (4 * 64) bits / 8 bits = 32
+            let inner_byte_slice =
+                std::slice::from_raw_parts_mut(inner_slice.as_mut_ptr() as *mut u8, 32);
+            let shortest_len = inner_byte_slice.len().min(bytes.len());
+            inner_byte_slice[..shortest_len].copy_from_slice(&bytes[..shortest_len]);
+        }
+        #[cfg(target_endian = "big")]
+        for word in self.0.iter_mut() {
+            *word = word.swap_bytes();
+        }
+    }
+
+    pub fn copy_to_le_byte_slice(&self, bytes: &mut [u8]) {
+        let inner_slice = self.0.as_slice();
+        unsafe {
+            // SAFETY SLICES are contiguous
+            // (4 * 64) bits / 8 bits = 32
+            let inner_byte_slice =
+                std::slice::from_raw_parts(inner_slice.as_ptr() as *const u8, 32);
+            let shortest_len = inner_byte_slice.len().min(bytes.len());
+            bytes[..shortest_len].copy_from_slice(&inner_byte_slice[..shortest_len]);
+        }
+        #[cfg(target_endian = "big")]
+        for sub_slice in bytes.chunks_mut(8) {
+            sub_slice.reverse();
+        }
```

Comment:
bytes can be a slice that does not have a lenght multiple of 8 right ? the las bytes still need some for of inversion I would believ 🤔

Reply:
added length checks